File: Baekjoon/Array/수열(2559).js
/**
 * 
 * 매일 측정한 기온
 * 연속적인 합이 가장 큰 값
 * 
 */


const rl = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout,
});

const input = [];

rl.on('line', (line) => {
    input.push(line);
});

rl.on('close', () => {
    const [N, K] = input[0].split(' ').map(Number);
    const temps = input[1].split(' ').map(Number);

    const sums = [0];
    for (let n=0; n<N; n++) {
        sums[n+1] = sums[n] + temps[n];
    }

    let MAX = Number.MIN_SAFE_INTEGER;
    for (let k=0; k<sums.length-K; k++) {
        const sum = sums[k+K] - sums[k];
        MAX = sum > MAX ? sum : MAX;
    }

    console.log(MAX);
});

File: Baekjoon/Array/수열(2559).py
import sys
input = sys.stdin.readline

N, K = map(int, input().split())
temps = list(map(int, input().split()))

sums = [0 for _ in range(N+1)]
for n in range(N):
    sums[n+1] = sums[n] + temps[n]

MAX = -sys.maxsize - 1
for k in range(len(sums)-K):
    sum = sums[k+K] - sums[k]
    MAX = sum if sum > MAX else MAX
print(MAX)